File: Sample/NendSDK_Sample/NativeAdViewController.m
//
//  NativeAdViewController.m
//  NendSDK_Sample
//
//  Copyright (c) 2015年 F@N Communications. All rights reserved.
//

#import "NativeAdViewController.h"

#import <NendAd/NADNativeClient.h>

@interface NativeAdViewController () <NADNativeDelegate>

@property (nonatomic, weak) IBOutlet UIActivityIndicatorView *indicator;
@property (nonatomic) NADNativeClient *client;

@end

@implementation NativeAdViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.

     // Xibから広告Viewを生成
    UINib *nib = [UINib nibWithNibName:self.nib bundle:nil];
    UIView<NADNativeViewRendering> *adView = [nib instantiateWithOwner:nil options:nil][0];
    adView.center = CGPointMake(CGRectGetMidX(self.view.bounds), CGRectGetMidY(self.view.bounds));
    adView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:adView];
    [self.view addConstraints:@[
        [NSLayoutConstraint constraintWithItem:adView
                                     attribute:NSLayoutAttributeCenterX
                                     relatedBy:NSLayoutRelationEqual
                                        toItem:self.view
                                     attribute:NSLayoutAttributeCenterX
                                    multiplier:1.f
                                      constant:0.f],
        [NSLayoutConstraint constraintWithItem:adView
                                     attribute:NSLayoutAttributeCenterY
                                     relatedBy:NSLayoutRelationEqual
                                        toItem:self.view
                                     attribute:NSLayoutAttributeCenterY
                                    multiplier:1.f
                                      constant:0.f],
        [NSLayoutConstraint constraintWithItem:adView
                                     attribute:NSLayoutAttributeWidth
                                     relatedBy:NSLayoutRelationEqual
                                        toItem:nil
                                     attribute:NSLayoutAttributeWidth
                                    multiplier:1.f
                                      constant:CGRectGetWidth(adView.bounds)],
        [NSLayoutConstraint constraintWithItem:adView
                                     attribute:NSLayoutAttributeHeight
                                     relatedBy:NSLayoutRelationEqual
                                        toItem:nil
                                     attribute:NSLayoutAttributeHeight
                                    multiplier:1.f
                                      constant:CGRectGetHeight(adView.bounds)],
    ]];

    [self.view bringSubviewToFront:self.indicator];

    [NADNativeLogger setLogLevel:NADNativeLogLevelWarn];
    
    self.client = [[NADNativeClient alloc] initWithSpotId:self.spotId apiKey:self.apiKey];
    
    __weak typeof(self) weakSelf = self;
    [self.client loadWithCompletionBlock:^(NADNative *ad, NSError *error) {
        if (ad) {
            ad.delegate = weakSelf;
            // 広告をViewに描画
            [ad intoView:adView advertisingExplicitly:NADNativeAdvertisingExplicitlyPR];
        } else {
            NSLog(@"load error: %@", error);
        }
        weakSelf.indicator.hidden = YES;
    }];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - NADNativeDelegate

- (void)nadNativeDidClickAd:(NADNative *)ad
{
    NSLog(@"nadNativeDidClickAd");
}

@end

File: Sample/NendSDK_Swift_Sample/NativeAdViewController.swift
//
//  NativeAdViewController.swift
//  NendSDK_Sample
//
//  Copyright (c) 2015年 F@N Communications. All rights reserved.
//

import UIKit
import NendAd

class NativeAdViewController: UIViewController, NADNativeDelegate {
    
    fileprivate var client: NADNativeClient?

    var spotId: String!
    var apiKey: String!
    var nib: String!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        self.title = "Simple"
        
        // Xibから広告Viewを生成
        let nib = UINib(nibName: self.nib, bundle: nil)
        if let adView = nib.instantiate(withOwner: nil, options: nil).first as? UIView {
            adView.center = CGPoint(x: self.view.bounds.midX, y: self.view.bounds.midY)
            adView.translatesAutoresizingMaskIntoConstraints = false
            self.view.addSubview(adView)
            let constraints = [
                NSLayoutConstraint(item: adView, attribute: .centerX, relatedBy: .equal, toItem: self.view, attribute: .centerX, multiplier: 1.0, constant: 0.0),
                NSLayoutConstraint(item: adView, attribute: .centerY, relatedBy: .equal, toItem: self.view, attribute: .centerY, multiplier: 1.0, constant: 0.0),
                NSLayoutConstraint(item: adView, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .width, multiplier: 1.0, constant: adView.bounds.width),
                NSLayoutConstraint(item: adView, attribute: .height, relatedBy: .equal, toItem: nil, attribute: .height, multiplier: 1.0, constant: adView.bounds.height)
            ]
            self.view.addConstraints(constraints)
            
            NADNativeLogger.setLogLevel(.warn)
            
            self.client = NADNativeClient(spotId: self.spotId, apiKey: self.apiKey)
            self.client!.load() { (ad, error) in
                if let nativeAd = ad {
                    nativeAd.delegate = self
                    // 広告をViewに描画
                    nativeAd.intoView(adView as! UIView & NADNativeViewRendering, advertisingExplicitly:.PR)
                } else {
                    print("error:\(error!)")
                }
            }
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - NADNativeDelegate
    func nadNativeDidClickAd(_ ad: NADNative!) {
        print(#function)
    }
}
